### ISSUE TITLE
Return fitted models after cv

### ISSUE THREAD

**Issue by vbugaevskii** [catboost/catboost#1524]
Hi!

There is no opportunity to get fitted models after cv, because catboost.cv returns only evaluation metric scores. At the same time popular ml libraries have such option in some kind.

For LightGBM there is optional argument [`return_cvbooster`](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.cv.html):

```python
cv = lgb.cv(params, X_train, show_stdv=False, stratified=True, verbose_eval=True,
            num_boost_round=trees, nfold=5, early_stopping_rounds=50,
            return_cvbooster=True, seed=72, )
```

XGBoost support such opportunity by using callbacks:

```python
cvbooster = []

def save_best_model(env):
    global cvbooster
    cvbooster[:] = [cvpack.bst for cvpack in env.cvfolds]

cv = xgb.cv(params, X_train, stratified=True, verbose_eval=True,
            num_boost_round=trees, nfold=5, early_stopping_rounds=50,
            seed=72, callbacks=[save_best_model, ])
```

Final models fitted on cv can be used in bagging ensemble. Also having individual models for each split will help investigating problems in data on specific fold.

It will be awesome to get such feature in future.

**Comment by vklyukin** [comment]
I will work on this🚀
